### PR TITLE
EDDF V1.0

### PR DESCRIPTION
**Pull request by Alpi-no**
I added in all the missing transitions for 07 and 25 L/C/R
@Fechulo  thank you for doing the most important ones!

### CODE REVIEW

**Comment by nielsvdweide** [comment]
Can we test this locally?

**Comment by Alpi-no** [comment]
As I constantly failed to fork a version for myself, I can't provide you with a demo :/

**Comment by Fechulo** [comment]
@jakobeng1303 I can add the changes to my own fork so that you can test it, however there seems to be a missing quotation mark on line 440.

**Comment by Alpi-no** [comment]
@Fechulo thanks out and sorry. That would be great!

**Comment by Alpi-no** [comment]
@Fechulo I would like to edit this airport again when the new routes are available to make this airport as realist (clearance limits -> no automatic clearance). Would you be so kind, to alter the airspace and terrain? Not know though. I myself will start working this EDDF v2.1 as soon, as the new options of routes are released.

**Comment by Fechulo** [comment]
@jakobeng1303 I found another 2 bugs on lines 557 and 505. But it's still not working, there's probably an extra comma somewhere, I'll keep looking
I'm not sure I understand what you mean by altering the airspace and terrain.

**Comment by nielsvdweide** [comment]
@jakobeng1303 To be fair, I quite like the automatic transitions... It gives me less workload...

**Comment by Alpi-no** [comment]
In real life, you would have a clearance limit at KERAX, Spessart, UNOKO or ROLIS and cleared out of there. At the moment, it's not your job to clear that but I think it should be! it would be the most realistic approach, however, it doesn't make sense to add this, until holds are included in routes, else planes would just overshoot the holdings. I hope that's more clear! 
@nielsvdweide I can understand that you like them but personally I think, theres not a lot to do without it, it's unrealistic and furthermore, I feel very limited

**Comment by nielsvdweide** [comment]
@jakobeng1303 yeah i know what you're saying :) It would be the most realistic approach, but then on the other hand (and this has been mentioned before) what are we trying to simulate? At the moment we are delivery/ground/tower and radar... 

As you said, i feel a clearance limit should be inserted with a plane holding at the end of it if no other clearance has been given...

**Comment by Alpi-no** [comment]
@nils at EDDF it might not play a big part but in other airports it does and I think we need some standards. Also in terms of size and jobs. I think airports like LOWW provide you with enough jobs and planes to handle comfortably without too much stress

**Comment by Fechulo** [comment]
@jakobeng1303 Do you mean expanding the airspace to include the clearance limits?
eddf
I just don't think it'd be very realistic. That's just my opinion though.

**Comment by nielsvdweide** [comment]
@jakobeng1303 I loved to play Schiphol, but with the new waypoints its just too much to handle... cant even see the runways anymore... But i know what you're saying :)

**Comment by Alpi-no** [comment]
@Fechulo basically, yes! 
@nielsvdweide is the version o EHAM you are talking about, the one in game? However I think, airports should be more or less difficult depending on their size!

**Comment by erikquinn** [comment]
@jakobeng1303 is this good to go?

**Comment by jakobeng1303** [comment]
As I don't have a fork for myself, I'm not able to test it but overall it should work. It might be good, however, to have a quick look at it 😬

**Comment by jakobeng1303** [comment]
Disregard my first comment. I will reread it tomorrow when I get back from school. I think that might be worth doing!

**Comment by Alpi-no** [comment]
There might be some mistakes but overall it should be fine. I would rework some parts tomorrow if you don't mind. I will include the standard clearance on one transition again, as it makes sense sort of

**Comment by erikquinn** [comment]
@jakobeng1303 No worries. I wouldn't really know what to look for though (other than it breaking altogether). So I'll just throw it up on my gh-pages, and give you a link to play around with it and make sure the changes are all working as expected. Stand by for that... :smile:

**Comment by jakobeng1303** [comment]
Looking at this again, despite some punctuation mistakes that might have happened, it's about as finished, as it gets at the current stage of the game

**Comment by Alpi-no** [comment]
Ok thank you!

**Comment by erikquinn** [comment]
Current version of the game: [erikquinn.github.io/atc](http://erikquinn.github.io/atc)
After this PR is merged: [erikquinn.github.io/atc/b/pr676](http://erikquinn.github.io/atc/b/pr676)

Let me know when you're finished what you find!

**Comment by Alpi-no** [comment]
I found one mistake where the ICAOSs name of the UNOKO25N is UNOKU25S instead of N. 
The game freezes and then crashes when I try to assign a transition like unoko.unoko25s.eddf

**Comment by Alpi-no** [comment]
Absolutely no clue why, especially because it happens in both versions

**Comment by Alpi-no** [comment]
it seems to be a problem with manually assigning the transition, as even reassigning a plane with the same transition, causes this problem. Assigning transitions in LOWW won't work either @erikquinn

**Comment by n8rzz** [comment]
do you see this same issue with any other airport or is this issue localized to this specific airport?

**Comment by Alpi-no** [comment]
I have the same issue with LOWW

**Comment by n8rzz** [comment]
are there any console errors?

**Comment by n8rzz** [comment]
better yet, what are the steps to reproduce the error you're seeing?

**Comment by Alpi-no** [comment]
due to my bad internet, uploading a picture doesn't work at the moment but it just freezes without any warnings
reproducing should be simple, go to EDDF or LOWW and tell any arriving plane (EDDF) kerax.kerax25n.eddf or (LOWW) mabod.mabod4k.loww

**Comment by erikquinn** [comment]
That should be either `rr KERAX.KERAX25N.EDDF` or `star KERAX25N`

**Comment by n8rzz** [comment]
when running `rr KERAX.KERAX25N.EDDF` this looks a legit bug with both v3 and v3.1 code.  It looks to be because `KERAX` is not listed as an entry point for the star. _as a note, v3.1 is local to my fork and is still very much in progress.  It contains a complete refactor of how fixes, sids and stars are handled._ 

#### v3.0.0:

screen shot 2016-11-15 at 5 18 23 pm

#### v3.1.0

screen shot 2016-11-15 at 5 19 59 pm

In the v3 code, the `getSTAR()` method is relatively untouched, so this looks like it was an issue before that nobody had run into.  Actually, I'm not sure if this _is_ an issue, I defer to @erikquinn here.  Can a star be used this way?  I would think that it could, thus issue, but I don't know for sure.

I'm still looking into what the problem is for `LOWW`, it might be the same but could be different.

**Comment by n8rzz** [comment]
I'm not able to reproduce any issue with either v3 or v3.1.0 at `LOWW`

File: assets/airports/eddf.json
Comment (erikquinn) [inline assets/airports/eddf.json:529]: `T173` is not a fix, which is what the game expects it to be. It is an airway, but it is not defined as an airway either, so the game really doesn't know what to make of it. When it is told there is an "entry point" of "T173", it tells the plane to fly to "T173" and then proceed along the `KERAX25N` star until you reach `DF421`.

So this should be amended to the first point, where the aircraft is to join the STAR route, which would be `GAPLA` in this case.

``` javascript
"entryPoints": {
  "GAPLA": ["GAPLA", ["DEMAB", "A160-"], ["SOLVU", "A130-"]]
},
```

This will need to be corrected throughout the STARs in order for them to be interpreted correctly by the sim.

Comment (erikquinn) [reply assets/airports/eddf.json:529]: Additionally, all the entries under `arrivals` will need to be amended, since they contain an invalid requested routing of `T180.UNOKO25S.EDDF` / `T150.ROLIS25N.EDDF` / `T173.KERAX25N.EDDF` / `T163.PSA25S.EDDF`.

Comment (erikquinn) [reply assets/airports/eddf.json:529]: Created #730 as there are some other issues with this, but the above should resolve the problems here. And in the meantime, `rr GAPLA.KERAX25N.EDDF` is the best bet for getting it to work properly. The `star` command should be fixed soon though.

**Review by erikquinn — CHANGES_REQUESTED** [review]
There are a few changes here that are needed for this to work. Let me know if you need any assistance with addressing them!

**Comment by Alpi-no** [comment]
I will review the changes and after doing that, will open a new request to keep things clean and simple. The one thing I'm not sure about is, that these are not things I changed, so it must have worked before! The only thing I did was, to add the missing STARS. That's also why I was surprised to not be able to use them!

**Comment by Alpi-no** [comment]
in #733 I have included all the changes

**Comment by Fechulo** [comment]
@jakobeng1303 I was the one who messed the transition names up. I didn't know that it could break the game and I didn't test it since players don't normally mess around with the STAR's

**Pull request closed without merging**
